### ISSUE TITLE
Build Fail on AMD

### ISSUE THREAD

**Issue by avildema** [openucx/ucx#6865]
### Describe the bug
Fail to build on AMD ubuntu

### Steps to Reproduce
```sh
./autogen.sh
../contrib/configure-release --prefix=/scrap/azure/agent-06/AZP_WORKSPACE/1/s/install --with-mpi
make -j2 install
```

### Setup and versions
swx-rdmz-instinct02

- Ubuntu 20.04
- x86_64
- 5.4.0-48-generic
- AMD EPYC 7702P 64-Core Processor

### Additional information (depending on the issue)

https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=18719&view=logs&j=f97760b2-5332-5514-7a21-2a026a1217cc&t=78fb67b3-8376-5236-7d82-1487b2abec27

raw log: https://dev.azure.com/ucfconsort/0b36e3f0-8ab9-4a48-b68b-4b2350e02c88/_apis/build/builds/18719/logs/84

```
/scrap/azure/agent-06/AZP_WORKSPACE/1/s/contrib/../src/uct/rocm/copy/rocm_copy_md.c: In function 'uct_rocm_copy_mem_dereg':
/scrap/azure/agent-06/AZP_WORKSPACE/1/s/contrib/../src/uct/rocm/copy/rocm_copy_md.c:162:18: error: redeclaration of 'status' with no linkage
  162 |     hsa_status_t status;
      |                  ^~~~~~
/scrap/azure/agent-06/AZP_WORKSPACE/1/s/contrib/../src/uct/rocm/copy/rocm_copy_md.c:156:18: note: previous declaration of 'status' was here
  156 |     hsa_status_t status;
      |                  ^~~~~~
/scrap/azure/agent-06/AZP_WORKSPACE/1/s/contrib/../src/uct/rocm/copy/rocm_copy_md.c:162:5: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
  162 |     hsa_status_t status;
      |     ^~~~~~~~~~~~
/scrap/azure/agent-06/AZP_WORKSPACE/1/s/contrib/../src/uct/rocm/copy/rocm_copy_md.c: In function 'uct_rocm_copy_rcache_mem_dereg_cb':
/scrap/azure/agent-06/AZP_WORKSPACE/1/s/contrib/../src/uct/rocm/copy/rocm_copy_md.c:273:47: error: passing argument 2 of 'uct_rocm_copy_mem_dereg' from incompatible pointer type [-Werror=incompatible-pointer-types]
  273 |     (void)uct_rocm_copy_mem_dereg(&md->super, &region->memh);
      |                                               ^~~~~~~~~~~~~
      |                                               |
      |                                               uct_rocm_copy_mem_t * {aka struct uct_rocm_copy_mem *}
/scrap/azure/agent-06/AZP_WORKSPACE/1/s/contrib/../src/uct/rocm/copy/rocm_copy_md.c:152:58: note: expected 'const uct_md_mem_dereg_params_t *' {aka 'const struct uct_md_mem_dereg_params *'} but argument is of type 'uct_rocm_copy_mem_t *' {aka 'struct uct_rocm_copy_mem *'}
  152 |                         const uct_md_mem_dereg_params_t *params)
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
cc1: all warnings being treated as errors
make[3]: *** [Makefile:749: copy/libuct_rocm_la-rocm_copy_md.lo] Error 1
```

```
2021-05-27T09:32:52.2197873Z configure: =========================================================
2021-05-27T09:32:52.2204303Z configure: UCX build configuration:
2021-05-27T09:32:52.2206111Z configure:       Build prefix:   /scrap/azure/agent-06/AZP_WORKSPACE/1/s/install
2021-05-27T09:32:52.2209346Z configure:  Configuration dir:   ${prefix}/etc/ucx
2021-05-27T09:32:52.2211163Z configure: Preprocessor flags:   -DCPU_FLAGS="" -I${abs_top_srcdir}/src -I${abs_top_builddir} -I${abs_top_builddir}/src
2021-05-27T09:32:52.2214911Z configure:         C compiler:   gcc -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch -Wno-pointer-sign -Werror-implicit-function-declaration -Wno-format-zero-length -Wnested-externs -Wshadow -Werror=declaration-after-statement
2021-05-27T09:32:52.2218208Z configure:       C++ compiler:   g++ -O3 -g -Wall -Werror -funwind-tables -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-label -Wno-long-long -Wno-endif-labels -Wno-sign-compare -Wno-multichar -Wno-deprecated-declarations -Winvalid-pch
2021-05-27T09:32:52.2220186Z configure:       Multi-thread:   disabled
2021-05-27T09:32:52.2221094Z configure:       NUMA support:   enabled
2021-05-27T09:32:52.2221969Z configure:          MPI tests:   enabled
2021-05-27T09:32:52.2222815Z configure:        VFS support:   no
2021-05-27T09:32:52.2223603Z configure:      Devel headers:   no
2021-05-27T09:32:52.2281182Z configure:           Bindings:   < >
2021-05-27T09:32:52.2337290Z configure:        UCS modules:   < >
2021-05-27T09:32:52.2395937Z configure:        UCT modules:   < ib rdmacm rocm cma knem >
2021-05-27T09:32:52.2453795Z configure:       CUDA modules:   < >
2021-05-27T09:32:52.2511573Z configure:       ROCM modules:   < >
2021-05-27T09:32:52.2567534Z configure:         IB modules:   < >
2021-05-27T09:32:52.2620552Z configure:        UCM modules:   < rocm >
2021-05-27T09:32:52.2676841Z configure:       Perf modules:   < rocm >
2021-05-27T09:32:52.2683890Z configure: =========================================================
```

**Comment by souravzzz** [comment]
Fixed in #6851

**Comment by souravzzz** [comment]
@avildema Can you please close the ticket? It has been fixed in #6851

**Comment by avildema** [comment]
Fixed
